src/go/mock-wsl/mock-wsl.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"golang.org/x/exp/slices"
+	"golang.org/x/text/encoding/unicode"
+)
+
+const (
+	ModeSequential = "sequential"
+	ModeRepeated   = "repeated"
+	ModeDefault    = ""
+)
+
+type commandEntry struct {
+	Args    []string `json:"args"`
+	Mode    string   `json:"mode,omitempty"`
+	Stdout  string   `json:"stdout,omitempty"`
+	Stderr  string   `json:"stderr,omitempty"`
+	UTF16LE bool     `json:"utf16le,omitempty"`
+	Code    int      `json:"code,omitempty"`
+}
+
+type configStruct struct {
+	Commands []commandEntry `json:"commands"`
+	Results  []bool         `json:"results,omitempty"`
+	Errors   []string       `json:"errors,omitempty"`
+}
+
+func writeFile(file *os.File, config *configStruct, errFmt string, v ...any) {
+	errString := ""
+	if errFmt != "" {
+		errString = fmt.Sprintf(errFmt, v...)
+		config.Errors = append(config.Errors, errString)
```

Comment:
If `errString` is non-empty, then we'll do a `log.Fatalf` at line 54, which shuts things down, right? So why add to `config.Errors`? It's not going to actually be used, right?

Reply:
Two days in a row I forget how Go often does I/O. Noted.